x-pack/plugins/endpoint/common/types.ts

```diff
@@ -336,3 +338,175 @@ export type ResolverEvent = EndpointEvent | LegacyEndpointEvent;
+    ),
+    before: schema.maybe(
+      schema.arrayOf(schema.string(), {
+        minSize: 2,
+        maxSize: 2,
+      }) as kbnConfigSchemaTypes.Type<[string, string]>
+    ),
+    sort: schema.maybe(schema.string()),
+    order: schema.maybe(schema.oneOf([schema.literal('asc'), schema.literal('desc')])),
+    query: schema.maybe(
+      schema.string({
+        validate(value) {
+          try {
+            fromKueryExpression(value);
+          } catch (err) {
+            return i18n.translate('xpack.endpoint.alerts.errors.bad_kql', {
+              defaultMessage: 'must be valid KQL',
+            });
+          }
+        },
+      })
+    ),
+
+    // rison-encoded string
+    filters: schema.maybe(
+      schema.string({
+        validate(value) {
+          try {
+            decode(value);
+          } catch (err) {
+            return i18n.translate('xpack.endpoint.alerts.errors.bad_rison', {
+              defaultMessage: 'must be a valid rison-encoded string',
+            });
+          }
+        },
+      })
+    ),
+
+    // rison-encoded string
+    date_range: schema.maybe(
+      schema.string({
+        validate(value) {
+          try {
+            decode(value);
+          } catch (err) {
+            return i18n.translate('xpack.endpoint.alerts.errors.bad_rison', {
+              defaultMessage: 'must be a valid rison-encoded string',
+            });
+          }
+        },
+      })
+    ),
+  },
+  {
+    validate(value) {
+      if (value.after !== undefined && value.page_index !== undefined) {
+        return i18n.translate('xpack.endpoint.alerts.errors.page_index_cannot_be_used_with_after', {
+          defaultMessage: '[page_index] cannot be used with [after]',
+        });
+      }
+      if (value.before !== undefined && value.page_index !== undefined) {
+        return i18n.translate(
+          'xpack.endpoint.alerts.errors.page_index_cannot_be_used_with_before',
+          {
+            defaultMessage: '[page_index] cannot be used with [before]',
+          }
+        );
+      }
+      if (value.before !== undefined && value.after !== undefined) {
+        return i18n.translate('xpack.endpoint.alerts.errors.before_cannot_be_used_with_after', {
+          defaultMessage: '[before] cannot be used with [after]',
+        });
+      }
+      if (
+        value.before !== undefined &&
+        value.sort !== undefined &&
+        value.sort !== EndpointAppConstants.ALERT_LIST_DEFAULT_SORT
+      ) {
+        return i18n.translate(
+          'xpack.endpoint.alerts.errors.before_cannot_be_used_with_custom_sort',
+          {
+            defaultMessage: '[before] cannot be used with custom sort',
+          }
+        );
+      }
+    },
+  }
+);
+
+/**
+ * Like TypeOf, but provides a type for creating the value that will match.
+ * schema.number accepts a string, so this allows strings for number
+ * schema.maybe creates an optional type, if such a type is a prop on an schema.object,
```

Comment:
Is it technically accurate to say "optional type" here? I see that it's clarified on the line below, but this comment seems a little ambiguous.

Reply:
reworded. Hopefully more clear?
```
/**
 * Takes a @kbn/config-schema 'schema' type and returns a type that represents valid inputs.
 * Similar to `TypeOf`, but allows strings as input for `schema.number()` (which is inline
 * with the behavior of the validator.) Also, for `schema.object`, when a value is a `schema.maybe`
 * the key will be marked optional (via `?`) so that you can omit keys for optional values.
 *
 * Use this when creating a value that will be passed to the schema.
 * e.g.
 * ```ts
 * const input: KbnConfigSchemaInputTypeOf<typeof schema> = value
 * schema.validate(input) // should be valid
 * ```
 */
```